k8s/controllers/actions/catalogeval_controller.go

```diff
@@ -0,0 +1,152 @@
+	// ApiClient is the client for Symphony API
+	ApiClient utils.ApiClient
+}
+
+//+kubebuilder:rbac:groups=federation.symphony,resources=catalogevalexpression,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=federation.symphony,resources=catalogevalexpression/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=federation.symphony,resources=catalogevalexpression/finalizers,verbs=update
+
+func (r *CatalogEvalReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ret ctrl.Result, reterr error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("Entering the CatalogEvalExpression reconciler")
+	evalCR := &federationv1.CatalogEvalExpression{}
+
+	err := r.Get(ctx, req.NamespacedName, evalCR)
+	if err != nil {
+		log.Error(err, "failed to parse CatalogEvalExpression")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	operationId := evalCR.GetOperationID()
+
+	// Clean up the cr if it is older than 10 hrs
+	if CheckNeedtoDelete(evalCR.ObjectMeta) {
+		// Delete the CR
+		log.Info("Deleting the CatalogEvalExpression as it is older than 10 hrs", "CR", req.NamespacedName)
+		if err := r.Delete(ctx, evalCR); err != nil {
+			// If delete fails, log the error and return (it shall retry)
+			log.Error(err, "Error deleting the CatalogEvalExpression", "CR", req.NamespacedName)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if evalCR.DeletionTimestamp.IsZero() {
+		// Get parent catalog
+		catalog := &federationv1.Catalog{}
+		if evalCR.Spec.ResourceRef.Name == "" || evalCR.Spec.ResourceRef.Namespace == "" {
+			log.Error(errors.New("ParentRef is not set"), "ParentRef is not set")
+			// Update status with results
+			status := &federationv1.ActionStatusBase{}
+			status.ActionStatus = federationv1.ActionResult{
+				Error: &federationv1.ProvisioningError{
+					Message: "ParentRef is not set",
+					Code:    "ParentRefNotSet",
+				},
+				Status:      federationv1.FailedActionState,
+				OperationID: operationId,
+			}
+			evalCR.Status = federationv1.CatalogEvalExpressionStatus{
+				ActionStatusBase: *status,
+			}
+			err = r.Update(ctx, evalCR)
+			if err != nil {
+				// if update fail, log the error and return (it shall retry)
+				log.Error(err, "Update failed", "CR", req.NamespacedName)
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, nil
+		}
+		if err := r.Get(ctx, *evalCR.Spec.ResourceRef.GetNamespacedName(), catalog); err != nil {
+			log.Error(err, "Error getting the parent catalog resource", "CR", req.NamespacedName)
+			status := &federationv1.ActionStatusBase{}
+			status.ActionStatus = federationv1.ActionResult{
+				Error: &federationv1.ProvisioningError{
+					Message: "ParentRef is not found",
+					Code:    "ParentRefNotFound",
+				},
+				Status:      federationv1.FailedActionState,
+				OperationID: operationId,
+			}
+			evalCR.Status = federationv1.CatalogEvalExpressionStatus{
+				ActionStatusBase: *status,
+			}
+			err = r.Update(ctx, evalCR)
```

Comment:
Will this encounter the reconcile loop again? https://github.com/eclipse-symphony/symphony/blob/578a5f9579280a355f3d4dd1b781d0116fe981a5/k8s/reconcilers/deployment.go#L593 In target / instance controller, we use sub status object update, instead of whole object update.